src/types.ts

```diff
@@ -35,8 +35,17 @@ export interface Props {
 
   // If users can toggle summarization, whether the toggle should be enabled by defualt.
   isSummaryToggleInitiallyEnabled?: boolean;
+
+  // The custom reranker configuration to use.
+  // For more information, see: https://docs.vectara.com/docs/api-reference/search-apis/reranking
+  rerankingConfiguration?: RerankingConfiguration;
 }
 
+export type RerankingConfiguration = {
+  rerankerId: number;
+  [key: string]: unknown;
```

Comment:
Can we make this type more specific?

```ts
export type RerankingConfiguration = {
  rerankerId: number;
  mmrConfig?: {
    diversityBias: number;
  };
}
```

Reply:
Does mmrConfig apply to all rerankers? I also purposely kept it open so we're ready to accept all kinds of knobs, and just document what each one does as it becomes available.

Reply:
`mmrConfig` only applies to the MMR reranker. It seems like it'd be trivial to keep this type updated with new knobs as they become available, at the same time as we update the docs?